rest-client/impl/impl-base/src/main/java/org/jboss/arquillian/extension/rest/client/BaseRestEnricher.java

```diff
@@ -89,25 +86,25 @@ private void addHeaders(Map<String, String> headersMap, Header annotation)
         }
     }
 
-    protected boolean allInSameContext(List<Servlet> servlets)
-    {
-        Set<String> context = new HashSet<String>();
-        for (Servlet servlet : servlets) {
-            context.add(servlet.getContextRoot());
-        }
-        return context.size() == 1;
-    }
-
     protected abstract Object enrichByType(Class<?> clazz, Method method, ArquillianResteasyResource annotation, Consumes consumes, Produces produces);
 
-    // Currently no way to share @ArquillianResource URL (URLResourceProvider) logic internally, copied logic
+    // Currently no way to request value of URI, need to manually invoke the lookup
     protected URI getBaseURL()
     {
-        HTTPContext context = metaDataInst.get().getContext(HTTPContext.class);
-        if (allInSameContext(context.getServlets())) {
-            return context.getServlets().get(0).getBaseURI();
+        Collection<ResourceProvider> resourceProviders = loader.get().all(ResourceProvider.class);
+        for(ResourceProvider resourceProvider: resourceProviders)
+        {
+            if(resourceProvider.canProvide(URI.class))
```

Comment:
What if we can have more than one provider? This will fail on the first one returning `null` but maybe there is another one which can pass? Just speculating

Reply:
@bartoszmajsak I read through those (I think I read them before).  It's not clear to me based on the comments in that thread what I need to change here.